### PR TITLE
1560 - Apply area and address specific rental property rules when creating applicant in internal portal

### PR DESCRIPTION
**Pull request by momentiris**
- Adds public endpoints for property and residential area rental rules validation
- Gathers some adapter imports under `import * as ...` instead for easy domain association and tidyness

### CODE REVIEW

File: src/adapters/leasing-adapter.ts [hunk 1]
```diff
@@ -132,6 +132,23 @@ const getContactByContactCode = async (
   }
 }
 
+const getTenantByContactCode = async (
+  contactCode: string
+): Promise<AdapterResult<any, unknown>> => {
```
Comment (driatic) [inline src/adapters/leasing-adapter.ts:137]: Let's type the success to a Tenant type here? Made a similar comment in the leasing PR :)

File: src/adapters/leasing-adapter.ts [hunk 2]
```diff
@@ -132,6 +132,23 @@ const getContactByContactCode = async (
...
+  try {
+    const res = await axios.get(
+      `${tenantsLeasesServiceUrl}/tenant/contactCode/${contactCode}`
+    )
+
+    if (!res.data.data) return { ok: false, err: 'not-found' }
+
+    return { ok: true, data: res.data.data }
```
Comment (driatic) [inline src/adapters/leasing-adapter.ts:145]: Same remark as in the leasing PR, let's use the response structure in https://github.com/Bostads-AB-Mimer/onecore-core/commit/04f673d7f450ee396c416d948dc7c5ea6fe20a6a

File: src/processes/parkingspaces/utils.ts [hunk 3]
```diff
@@ -10,3 +11,49 @@ export const makeProcessError = <E = any>(
...
   response,
 })
+
+type ValidationResult =
+  | Awaited<ReturnType<typeof leasingAdapter.validatePropertyRentalRules>>
+  | Awaited<
+      ReturnType<typeof leasingAdapter.validateResidentialAreaRentalRules>
+    >
+
+export const validateRentalRules = (
```
Comment (driatic) [inline src/processes/parkingspaces/utils.ts:21]: 👍

File: src/services/lease-service/index.ts [hunk 4]
```diff
@@ -538,12 +545,238 @@ export const routes = (router: KoaRouter) => {
    *       - bearerAuth: []
    */
   router.get('/applicants/:contactCode', async (ctx) => {
```
Comment (driatic) [inline src/services/lease-service/index.ts:547]: Super well documented, nice work! 🔥

**Review by driatic — CHANGES_REQUESTED** [review]
Overall, really nice work! I hate to request changes when this has been such a big chore but I think we should export the tenant type and make the response structure align with main. Almost there! 💪

File: src/adapters/leasing-adapter.ts [hunk 5]
```diff
@@ -132,6 +132,23 @@ const getContactByContactCode = async (
   }
 }
 
+const getTenantByContactCode = async (
+  contactCode: string
+): Promise<AdapterResult<any, unknown>> => {
+  try {
+    const res = await axios.get(
+      `${tenantsLeasesServiceUrl}/tenant/contactCode/${contactCode}`
```
Comment (driatic) [inline src/adapters/leasing-adapter.ts:140]: `tenants` to align with DIGG-standard

File: src/adapters/leasing-adapter.ts (hunk 1 above)
Comment (momentiris) [reply src/adapters/leasing-adapter.ts:137]: Will use after onecore-types/1560 is merged

File: src/adapters/leasing-adapter.ts (hunk 5 above)
Comment (momentiris) [reply src/adapters/leasing-adapter.ts]: https://github.com/Bostads-AB-Mimer/onecore-core/pull/125/commits/d4edf8ca7d68f41eddd72fc5c31daef74786150b

File: src/adapters/leasing-adapter.ts (hunk 2 above)
Comment (momentiris) [reply src/adapters/leasing-adapter.ts]: https://github.com/Bostads-AB-Mimer/onecore-core/pull/125/commits/d4edf8ca7d68f41eddd72fc5c31daef74786150b

File: src/adapters/leasing-adapter.ts (hunk 1 above)
Comment (momentiris) [reply src/adapters/leasing-adapter.ts]: https://github.com/Bostads-AB-Mimer/onecore-core/pull/125/commits/c0c6d9245ba3a8636fd981711a14781cc4a0ae54

**Review by driatic — APPROVED** [review]
Super!

**Pull request merged**